docs/define_a_subquery.md

```diff
@@ -85,16 +85,113 @@ filter:
 
 ## The GraphQL Schema
 
+### Defining Entities
 It is must define the GraphQL schemas inside of `schema.graphql` file. To know how to write in  "GraphQL schema language",
-we recommend to check out on [Schemas and Types](https://graphql.org/learn/schema/#type-language).
+we recommend checking out on [Schemas and Types](https://graphql.org/learn/schema/#type-language).
 
-We currently supporting flowing types:
+
+### Optional and required fields
+Each entity must define its required field `id` with the type of `ID!`, it used as the primary key and unique among all entities of the same type.
+
+**Required filed** in the entity are indicated by the `!`. 
+
+Please see the example below:
+
+```graphql
+type Example @entity {
+  id: ID! #id Entity is always required
+  name: String! # This is a required field
+  address: String #This is an optional field
+}
+```
+
+### Supported scalars and types
+
+We currently supporting flowing scalars:
 - `ID`
 - `Int`
 - `String`
 - `BigInt`
 - `Date`
+- `Boolean`
+
+For nested relationship entities, you might use the defined entity's name as one of the fields. Please see in [Entity Relationships](#entity-relationships).
+
+### Entity Relationships
+
+An entity often has nested relationships with other entities. Set the field value to another entity name will define a unidirectional relationship between these two entities by default.
+
+From the entity's relationship perspectives, the entity can `belongsTo`,`hasOne`or`hasMany` with other entities. In following examples covers these scenarios.
+
+#### One-to-One relationships
+
+Define a Passport entity type with a required one-to-one relationship with a Person entity type:
+
+```graphql
+type Person @entity {
+  id: ID!
+}
+
+type Passport @entity {
+  id: ID!
+  owner: Person!
+}
+```
+
+#### One-to-Many relationships
 
+Define a Contact entity type with an optional one-to-many relationship with a Person entity type.
+And remember, use bracket to indicate a field type is multiple entities.
+
+```graphql
+type Person @entity {
+  id: ID!
+  contacts: [Contact] 
+}
+
+type Contact @entity {
+  id: ID!
+  phone: String!
+}
+```
+
+#### Many-to-Many relationships
+
+Many-to-Many relationships can be implement by pointing to the same entity name in multiple fields of another entity.
```

Comment:
this is a special case for many to many, when both side is the same entity. but there are other cases as well.

I think the key point is a middle entity to connect both side